This is an attempt to make GHC build reproducible. The name of .c files may end
up in the resulting binary (in the debug section), but not the directory.

Since the temporary directory already contains the process id, there is no need
to put it into the filename as well.

Index: ghc-7.10.1/compiler/main/SysTools.hs
===================================================================
--- ghc-7.10.1.orig/compiler/main/SysTools.hs	2015-03-21 23:30:07.000000000 +0100
+++ ghc-7.10.1/compiler/main/SysTools.hs	2015-05-24 10:45:40.537513384 +0200
@@ -1072,8 +1072,7 @@
 newTempName :: DynFlags -> Suffix -> IO FilePath
 newTempName dflags extn
   = do d <- getTempDir dflags
-       x <- getProcessID
-       findTempName (d </> "ghc" ++ show x ++ "_")
+       findTempName (d </> "ghc_")
   where
     findTempName :: FilePath -> IO FilePath
     findTempName prefix
@@ -1088,8 +1087,7 @@
 newTempLibName :: DynFlags -> Suffix -> IO (FilePath, FilePath, String)
 newTempLibName dflags extn
   = do d <- getTempDir dflags
-       x <- getProcessID
-       findTempName d ("ghc" ++ show x ++ "_")
+       findTempName d ("ghc_")
   where
     findTempName :: FilePath -> String -> IO (FilePath, FilePath, String)
     findTempName dir prefix
